tests/bluetooth/bsim/mesh/src/test_dfu.c

```diff
@@ -764,12 +881,168 @@ static void test_cli_fail_on_persistency(void)
 	PASS();
 }
 
+static int blob_id_set(const char *name, size_t len_rd,
+		       settings_read_cb read_cb, void *cb_arg)
+{
+	ssize_t len;
+	uint64_t blob_id_read;
+
+	len = read_cb(cb_arg, &blob_id_read, 4);
+	if (len < 0) {
+		LOG_ERR("Failed to read value (err %zd)", len);
+		return len;
+	}
+
+	blob_id = blob_id_read;
+	return 0;
+}
+
+SETTINGS_STATIC_HANDLER_DEFINE(blob_id, "blob_id", NULL, blob_id_set, NULL, NULL);
```

Comment:
Personally, I like leaving blob id randomization in `bt_mesh_dfu_cli_send` and using undefined value to tell it that it should be random, or passed by user otherwise. I think 0xFFFFFFFF is good marker for this. Then this will be defined and passed in DFD also. IMO this has advantage that user has also way to use random one if magic value is used. Of course user could use `sys_rand_get` oneself, but it would make things simpler

Reply:
Ok, but then I'd suggest to use `0` as uninitialized value so that user can omit this field when initializing `struct bt_mesh_dfu_cli_xfer`. It should also be fixed in `dfd_srv.c` as currently it will have a garbage: https://github.com/zephyrproject-rtos/zephyr/blob/169270e911d3d3c699453ab0fa8618363f953913/subsys/bluetooth/mesh/dfd_srv.c#L841.

Reply:
ah, you want this to be part of xfer, not another argument to `bt_mesh_dfu_cli_send`. I was thinking about something like:
```c
int bt_mesh_dfu_cli_send(struct bt_mesh_dfu_cli *cli,
			 const struct bt_mesh_blob_cli_inputs *inputs,
			 const struct bt_mesh_blob_io *io,
			 const struct bt_mesh_dfu_cli_xfer *xfer,
			 uint64_t id)
{
	struct bt_mesh_dfu_target *target;

	if (bt_mesh_dfu_cli_is_busy(cli)) {
		return -EBUSY;
	}

	cli->xfer.blob.mode = xfer->mode;
	cli->xfer.blob.size = xfer->slot->size;

	if (id == BT_MESH_BLOB_ID_UNDEFINED) {
		sys_rand_get(&cli->xfer.blob.id, sizeof(cli->xfer.blob.id));
	} else {
		cli->xfer.blob.id = id;
	}
```
but I can see adding this to xfer makes it more compact. Then yes, it should be 0